### ISSUE TITLE
The examples for creating a node should include creating a node from a template

### ISSUE THREAD

**Issue by 1001QAdotNET** [GNS3/gns3-server#1791]
We rarely create a node using the method shown in the examples
This should be documented and shown explicitely
http://api.gns3.net/en/2.2/api/v2/controller/template/projectsprojectidtemplatestemplateid.html

**Comment by b-ehlers** [comment]
I agree, that more examples of the API would be helpful.

But for me the most instructive way is to watch the GUI using the API. I'm using wireshark for that, for details see
<https://www.b-ehlers.de/blog/posts/2019-08-12-gns3-api-tips-tricks/#watching-the-api-usage-of-the-gns3-gui>.

Here a wireshark trace, when creating a node from a template:

![wireshark_packetlist](https://user-images.githubusercontent.com/49040325/85922541-72d55980-b884-11ea-914a-0d2ed9f076c5.png)
![wireshark_request](https://user-images.githubusercontent.com/49040325/85922546-78cb3a80-b884-11ea-8667-1449bb6154b3.png)

**Comment by 1001QAdotNET** [comment]
you can watch what happens on the console as well.
I too flirted with the idea but then I noticed that for instance when it is creating a node there is a couple of steps that happen in the sequence, multiple API calls which do not seem to match the API call. (I created a node in the gui and then I checked the console ...what should have been a single call was done via a couple of APIs calls. maybe that is what the controller does in background

**Comment by grossmj** [comment]
Agreed, the documentation should include more examples. We have started working on a new API documentation which should be much better.
